keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -32,218 +31,105 @@ status: provisional
-
-#### Trace Utility Package
-
-This package will be able to create spans from the span context embedded in the `trace.kubernetes.io/context` object annotation, in addition to embedding context from spans back into the annotation. This package will facilitate propagating traces through kubernetes objects.  The exported functions include:
-
-```golang
-// InitializeExporter initializes the trace exporting service with the provided service name.
-// Components should use this initializer to ensure common behavior.
-func InitializeExporter(service string)
-
-// StartSpanFromObject constructs a new Span using the context attached to the object as the parent SpanContext.  It mirrors trace.StartSpan, but for kubernetes objects.
-func StartSpanFromObject(ctx context.Context, obj meta.Object, spanName string) (context.Context, *trace.Span, error)
-
-// EncodeContextIntoObject encodes the SpanContext contained in the context into the object
-func EncodeContextIntoObject(ctx context.Context, obj meta.Object)
-
-// RemoveSpanContextFromObject removes the SpanContext attached to an object, if one exists
-func RemoveSpanContextFromObject(obj meta.Object) 
-```
-
-#### Tracing Pod Lifecycle
+The API Server will use the [OpenTelemetry exporter format](https://github.com/open-telemetry/opentelemetry-proto), which exports traces to a local port.  This format is compatible with the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector), which allows importing and configuring exporters for trace storage backends to be done out-of-tree in addition to other useful features.  The exporter stores spans in memory, and uses the [batching processor](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#batching-processor) to batch requests and send them asynchronously.
 
-As we move forward with this KEP, we will use the aforementioned trace utility package to trace pod-related operations across the scheduler and kubelet. In code, this corresponds to creating a span (i.e. `ctx, span := trace.StartSpan(ctx, "Component.SampleSpan")`) at the beginning of an operation, and ending the span afterwards (`span.End()`). All calls to tracing functions will be gated with the `ObjectLifecycleTracing` alpha feature-gate, and will be disabled by default.
+Add configuration to the API Server required to configure the opentelemetry exporter, including the address and egress proxy to send spans to. The [egress proxy](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190226-network-proxy.md) can be added to the opentelemetry exporter by adding a ContextDialer grpc DialOption similar to the one used by the apiserver's etcd client.
```

Comment:
How is this configuration going in? Flag? File? Dynamic API?

Who usually configures such things? Developers or cluster admins?

Reply:
It would just require adding another egress config destination in https://github.com/kubernetes/kubernetes/blob/4b9b9ab75376b7b53876ab6b2be42d0940c7eb26/staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go#L53 in addition to "Master" "Cluster" "Etcd".

Although, we generally expect the OT collector to be run on the local network of the component, either as a sidecar, or on the host network if all master components are configured that way.

Reply:
Should we add the egress selector changes to the doc?

Reply:
added the egress selector config to the KEP.